components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v2/src/main/java/org/wso2/carbon/identity/rest/api/user/recovery/v2/impl/core/utils/RecoveryUtil.java

```diff
@@ -0,0 +1,576 @@
+                    return buildRequestNotAcceptableResponseObject(className, exception.getMessage(), errorCode,
+                            correlationId);
+                case RETRY_ERROR_CATEGORY:
+                    return buildRetryPasswordResetObject(className, tenantDomain, exception.getMessage(), errorCode,
+                            code, correlationId);
+                default:
+                    return buildConflictRequestResponseObject(className, exception.getMessage(), errorCode,
+                            correlationId);
+            }
+        } else {
+            return buildConflictRequestResponseObject(className, exception.getMessage(), errorCode, correlationId);
+        }
+    }
+
+    /**
+     * Logs the error, builds a internalServerErrorException with specified details and throws it.
+     *
+     * @param className     Class name
+     * @param message       Error message
+     * @param code          Error code
+     * @param correlationId Correlation Id
+     * @param throwable     Error
+     * @return WebApplicationException
+     */
+    public static WebApplicationException buildInternalServerErrorResponse(String className, String message,
+                                                                           String code, String correlationId,
+                                                                           Throwable throwable) {
+
+        if (StringUtils.isNotBlank(className)) {
+            message = String.format("%s : %s - %s", LOG_MESSAGE_PREFIX, className, message);
+        }
+        if (throwable == null) {
+            log.error(message);
+        } else {
+            log.error(message, throwable);
+        }
+        return buildInternalServerError(code, correlationId);
+    }
+
+    /**
+     * Build API call information.
+     *
+     * @param type   Type of the API call
+     * @param rel    API relation
+     * @param apiUrl Url of the API
+     * @param data   Additional data
+     * @return APICall {@link APICall} which encapsulates the API name and the url
+     */
+    public static APICall buildApiCall(String type, String rel, String apiUrl, String data) {
+
+        if (StringUtils.isNotEmpty(data)) {
+            apiUrl = String.format(apiUrl, data);
+        }
+        APICall apiCall = new APICall();
+        apiCall.setType(type);
+        apiCall.setRel(rel);
+        apiCall.setHref(apiUrl);
+        return apiCall;
+    }
+
+    /**
+     * @deprecated This was deprecated because the requirement is to get the absolute and relative URIs by using the
+     * {@link ServiceURLBuilder} methods.
+     * @since 1.1.6
+     *
+     * Please use {@link #buildURIForBody(String, String, String)} method to build URIs for body.
+     *
+     * Build the relative url.
+     * @param tenantDomain Tenant Domain
+     * @param endpoint     API endpoint
+     * @return Url
+     */
+    public static String buildUri(String tenantDomain, String endpoint, String baseUrl) {
```

Comment:
Why are we introducing a new deprecated method? What are the usages of this?

Reply:
Removed with e70ec1e4f6ae43fb67f7dea0c139494935bdd1ca